File: DataStructure/10845_큐.cpp
#include <iostream>
#include <queue>

using namespace std;

void solution(void) {
	int T;
	queue <int> q;
	
	int value;
	string cmm;
	
	cin >> T;
	for (int i=0; i<T;++i) {
		cin >> cmm;

		if (cmm == "push") {
			cin >> value;
			q.push(value);
		}

		else if (cmm == "pop") {
			if (q.empty())
				cout << -1 << '\n';
			else {
				cout << q.front() << '\n';
				q.pop();
			}
		}

		else if (cmm == "size") 
			cout << q.size() << '\n';
		
		else if (cmm == "empty") {
			cout << q.empty() << '\n';
		}

		else if (cmm == "front") {
			if (q.empty())
				cout << -1 << '\n';
			else 
				cout << q.front() << '\n';
			
		}
		
		else if (cmm == "back") {
			if (q.empty())
				cout << -1 << '\n';
			else 
				cout << q.back() << '\n';
			
		}
	}
	return;
}

int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	solution();
	return 0;
}

File: DataStructure/10845_큐.py
#큐
import queue
#입출력
from sys import stdin

#큐 선언
q=queue.Queue()
#입력 
input = stdin.readline

T=int(input())
for i in range(T):
    #명령어 입력
    comm,*n=input().split()

    #push
    if comm=='push':
        q.put(int(n[0]))
    
    #pop
    elif comm=='pop':
        print(q.get() if not q.empty() else -1)
    
    #size
    elif comm=='size':
        print(q.qsize())
    
    #empty
    elif comm=='empty':
        print(1 if q.empty() else 0)
    
    #front
    elif comm=='front':
        print(q.queue[0] if not q.empty() else -1)
    
    #back
    elif comm=='back':
        print(q.queue[-1] if not q.empty() else -1)